src/lib.rs

```diff
@@ -0,0 +1,1166 @@
+/// benchmarked your particular use case and are sure that a different storage type is superior.
+///
+///
+/// ## Migrating from `Vec<bool>`
+///
+/// As far as possible, `Vob` is intended to have a superset of `Vec<bool>`'s interface, which
+/// should make porting most code fairly simple. However, `Vec<bool>` contains several functions
+/// which are not yet implemented in `Vob`: these are missing simply due to a lack of a current
+/// use-case rather than because of any fundamental incompatibilities.
+///
+/// There is one missing feature which, currently, is impossible to implement: assignment to an
+/// index. In other words one cannot currently express `v[0] = true` for a `Vob` `v`. Until
+/// [`IndexGet` / `IndexMove` and equivalents](https://github.com/rust-lang/rfcs/issues/997) are
+/// implemented in `rustc`, this restriction appears to be inevitable. Note that referencing by
+/// index works (though via a hack identical to that used in `BitVec`): one can write
+/// `println!("{}", v[0])` for a `Vob` `v`, for example.
+///
+///
+/// ## Migrating from `BitVec`
+///
+/// `Vob` is directly inspired by `BitVec`, but aims to provide an interface aligned to `Vec<bool>`
+/// Several functions in `BitVec` have different names in `Vob`, but porting is in general fairly
+/// simple. The main semantic difference is that `Vob`s [`clear()`](struct.Vob.html#method.clear)
+/// function empties the `Vob` of contents (i.e. sets its length to 0), whereas `BitVec`s function
+/// of the same name unsets all bits (keeping the length unchanged). `Vob`s equivalent is
+/// [`set_all(false)`](struct.Vob.html#method.set_all).
+#[derive(Clone)]
+pub struct Vob<T=usize> {
+    /// How many bits are stored in this Vob?
+    bits: usize,
+    /// The underlying storage. Since the storage consists of (potentially multiple-byte) blocks,
+    /// there may be "unused" bits in the final block. We guarantee that, at all points visible to
+    /// the user, the "unused" bits are set to 0.
+    vec: Vec<T>
+}
+
+// In an ideal world, Rust's type defaults would allow us to fold the two impl blocks into one and
+// say "impl Vob<T: usize>", but currently that doesn't work.
+impl Vob<usize> {
+    /// Constructs a new, empty Vob (with `usize` backing storage, which is likely to be the best
+    /// choice in nearly all situations).
+    ///
+    /// The Vob will not allocate until elements are pushed onto it.
+    pub fn new() -> Vob<usize> {
+        Default::default()
+    }
+
+    /// Constructs a new, empty Vob (with `usize` backing storage, which is likely to be the best
+    /// choice in nearly all situations) with the specified capacity.
+    ///
+    /// The Vob will be able to hold at least `capacity` elements without reallocating. If
+    /// `capacity` is 0, the vector will not allocate.
+    pub fn with_capacity(capacity: usize) -> Vob<usize> {
+        Vob {
+            bits: 0,
+            vec: Vec::with_capacity(storage_size_align::<usize>(capacity))
```

Comment:
So you round up to the next alignment boundary? Why?

Reply:
Otherwise you won't have enough space to store the bits.

Reply:
Not sure I follow.

Suppose the use does:
```
let vob = vob::with_capacity(11);
```

Then you would actually allocate some larger amount than 11, yet 11 would have been enough?

Reply:
But you can't store 11 bits on their own: if you have 8 bit blocks, for example, you need two blocks to store 11 bits.